File: src/ts/Models/Entities/Genre.ts
import * as _ from 'lodash';

export default class Genre {
    private _id: string;
    private _name: string;
    private _uri: string;

    public get Id(): string {
        return this._id;
    }
    public get Name(): string {
        return this._name;
    }
    public get Uri(): string {
        return this._uri;
    }

    public constructor(name: string, uri: string) {
        if (!name || name === '')
            throw new Error('name is required.');
        if (!uri || uri === '')
            throw new Error('uri is required.');

        this._id = _.uniqueId('genre_');
        this._name = name;
        this._uri = uri;
    }
}

File: aspCore/Models/Genres/Genre.cs
using MusicFront.Models.Relations;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MusicFront.Models.Genres
{
    [Table("genres")]
    public class Genre
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Uri { get; set; }

        public List<GenreArtist> GenreArtists { get; set; }

        public List<GenreAlbum> GenreAlbums { get; set; }
    }
}
